### PR TITLE
Miscellaneous fixups

### PR DESCRIPTION
**Pull request by GiudGiud**
Things I ran into during Easter weekend MaCaw dev

### CODE REVIEW

File: framework/src/meshgenerators/PatternedMeshGenerator.C [hunk 1]
```diff
@@ -310,8 +310,8 @@ PatternedMeshGenerator::mergeSubdomainNameMaps(
     const auto name_to_insert = id_name_pair.second;
     if (main_subdomain_map_name_list.find(name_to_insert) != main_subdomain_map_name_list.end())
       paramError("inputs",
-                 "The input meshes both contain subdomain name " + name_to_insert +
-                     " that correspond to conflicting subdomain ids.");
+                 "Two of the input meshes contain a subdomain with then name '" + name_to_insert +
```
Comment (lindsayad) [inline framework/src/meshgenerators/PatternedMeshGenerator.C:313]:
```suggestion
                 "Two of the input meshes contain a subdomain with the name '" + name_to_insert +
```

File: framework/src/problems/FEProblemBase.C [hunk 2]
```diff
@@ -273,8 +273,11 @@ FEProblemBase::validParams()
   params.addParamNamesToGroup(
       "skip_nl_system_check kernel_coverage_check boundary_restricted_node_integrity_check "
       "boundary_restricted_elem_integrity_check material_coverage_check fv_bcs_integrity_check "
-      "material_dependency_check",
+      "material_dependency_check check_uo_aux_state error_on_jacobian_nonzero_reallocation",
       "Simulation checks");
+  params.addParamNamesToGroup("use_nonlinear previous_nl_solution_required nl_sys_names "
+                              "error_on_jacobian_nonzero_reallocation ignore_zeros_in_jacobian",
```
Comment (lindsayad) [inline framework/src/problems/FEProblemBase.C:279]: Is there any problem with having the same parameter in two different groups?

File: test/tests/mesh/save_in_meshes/tests
Comment (lindsayad) [inline test/tests/mesh/save_in_meshes/tests:1]: If modifying these details, can you add the commas and such to make the whole requirement plus details a "well-structured" sentence?

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: framework/src/problems/FEProblemBase.C (hunk 2 above)
Comment (lindsayad) [reply framework/src/problems/FEProblemBase.C:279]: (Referring to `error_on_jacobian_nonzero_reallocation`)

Comment (GiudGiud) [reply framework/src/problems/FEProblemBase.C:279]: seems it just gets ignored the second time. I ll keep it only once

File: test/tests/mesh/save_in_meshes/tests
Comment (GiudGiud) [reply test/tests/mesh/save_in_meshes/tests:1]: done

**Review by lindsayad — APPROVED** [review]

**Pull request merged**